pkgs/development/python-modules/bokeh/default.nix

```diff
@@ -51,55 +96,102 @@ buildPythonPackage rec {
     })
   ];
 
-  disabled = isPyPy || isPy27;
+  postPatch = ''
+    # workaround for setuptools-git-versioning not supporting overrides
+    substituteInPlace pyproject.toml \
+      --replace 'dynamic = ["version"]' 'version = "${version}"'
 
-  nativeBuildInputs = [
-    pythonImportsCheckHook
+    # inject fully built artifacts
+    rm -r src/bokeh/server/static
+    ln -s "${bokehjs}/lib/node_modules/@bokeh/bokehjs/build" "src/bokeh/server/static"
+    touch PKG-INFO
+  '';
+
+  propagatedBuildInputs = [
+    contourpy
+    jinja2
+    numpy
+    packaging
+    pandas
+    pillow
+    pyyaml
+    tornado
+    xyzservices
   ];
 
   pythonImportsCheck = [
     "bokeh"
   ];
 
   nativeCheckInputs = [
-    mock
-    pytest
-    pillow
-    selenium
-    pytz
-    flaky
-    networkx
     beautifulsoup4
-    requests
+    firefox
+    geckodriver
+    git
+    json5
     nbconvert
-    icalendar
+    nodejs
+    pytest-asyncio
+    pytest-html
+    pytest-rerunfailures
+    pytest-xdist
+    pytestCheckHook
+    requests
+    requests-unixsocket
+    selenium
   ];
 
-  propagatedBuildInputs = [
-    contourpy
-    pillow
-    jinja2
-    python-dateutil
-    six
-    pyyaml
-    tornado
-    numpy
-    pandas
-    packaging
-    typing-extensions
-    xyzservices
-  ]
-  ++ lib.optionals ( isPy27 ) [
-    futures
+  dontUsePytestXdist = true;
+
+  pytestFlagsArray = [
+    # limit parallelism, so we don't run into socket bind conflicts
+    "--numprocesses=$(( $NIX_BUILD_CORES > 4 ? 4 : $NIX_BUILD_CORES ))"
+    # chromium segfaults during tests
+    "--driver=firefox"
+    # retry some tests
+    "--reruns=3"
+  ];
+
+  disabledTestPaths = [
+    # tries to fetch node modules
+    "tests/test_bokehjs.py"
+    # expects to be executed in a git repository
+    "tests/test_defaults.py"
+    # fails to find examples
+    "tests/test_examples.py"
+    # linters
+    "tests/codebase"
+    # RuntimeError: bokeh sample data directory does not exist, please execute bokeh.sampledata.download
+    "tests/unit/bokeh/sampledata"
+    "tests/unit/bokeh/util/test_sampledata__util.py"
+  ];
+
+  disabledTests = [
+    # tries to fetch node modules
+    "test_ext_commands"
+    #  AssertionError: assert [{'extends': ...rop3'}], ...}] == [{'name': 'te...ro...
+    "test_serialization_data_models"
+    # Failed: Did not find pattern in process output
+    "test_websocket_max_message_size_printed_out"
+    # FileNotFoundError: [Errno 2] No such file or directory: '/build/source/tests/unit/bokeh/embed/latex_label/dist/latex_label.js
+    # FileNotFoundError: [Errno 2] No such file or directory: '/build/source/tests/unit/bokeh/embed/ext_package_no_main/dist/ext_package_no_main.js'
+    "test_with_CDN_resources"
+    "test_with_INLINE_resources"
+    # selenium.common.exceptions.TimeoutException: Message: Failed to read marionette port
+    "test_issue_11694"
+    "test_bkroot_added_to_target"
+    "test_includes_hidden_glyphs_by_default"
```

Comment:
Basically these exceptions are creeping up towards the end. I can't see what causes them.